components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -204,27 +225,32 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
 
       // Stopping means that the workspace is currently shutting down. It could go to stopped every moment.
       case "stopping":
+        if (isHeadless) {
+          return <HeadlessWorkspaceView workspaceId={this.state.workspaceInstance.workspaceId} />;
+        }
         phase = StartPhase.Stopping;
         statusMessage = <div>
           <div className="flex space-x-3 items-center rounded-xl m-auto px-4 h-16 w-72 mt-4 bg-gray-100">
             <div className="rounded-full w-3 h-3 text-sm bg-gitpod-kumquat">&nbsp;</div>
             <div>
               <p className="text-gray-700 font-semibold">{this.state.workspaceInstance.workspaceId}</p>
-              <p className="w-56 truncate">{this.state.contextUrl}</p>
+              <p className="w-56 truncate">{this.state.workspace?.contextURL}</p>
             </div>
           </div>
         </div>;
         break;
 
       // Stopped means the workspace ended regularly because it was shut down.
       case "stopped":
+        const pendingChanges = getPendingChanges(this.state.workspaceInstance);
         phase = StartPhase.Stopped;
         statusMessage = <div>
             <div className="flex space-x-3 items-center rounded-xl m-auto px-4 h-16 w-72 mt-4 bg-gray-100">
             <div className="rounded-full w-3 h-3 text-sm bg-gray-300">&nbsp;</div>
             <div>
               <p className="text-gray-700 font-semibold">{this.state.workspaceInstance.workspaceId}</p>
-              <p className="w-56 truncate">{this.state.contextUrl}</p>
+              <p>Pending changes: {pendingChanges.length}</p>
```

Comment:
Ah, I snuck this in (pending changes in Stopped screen) but then I should also make it look right.